src/uct/ib/mlx5/dv/ib_mlx5dv_md.c

```diff
@@ -170,31 +170,30 @@ static ucs_status_t uct_ib_mlx5_devx_reg_ksm_data_addr(
     return status;
 }
 
-static ucs_status_t
-uct_ib_mlx5_devx_reg_ksm_data_contig(uct_ib_mlx5_md_t *md,
-                                     uct_ib_mlx5_devx_mr_t *mr, off_t offset,
-                                     void *address, int atomic,
-                                     const char *reason,
-                                     struct mlx5dv_devx_obj **mr_p,
-                                     uint32_t *mkey)
+static ucs_status_t uct_ib_mlx5_devx_reg_ksm_data_contig(
+        uct_ib_mlx5_md_t *md, uct_ib_mlx5_devx_mr_t *mr, off_t offset,
+        void *address, void *iova, int atomic, const char *reason,
+        struct mlx5dv_devx_obj **mr_p, uint32_t *mkey)
 {
     size_t mr_length = mr->super.ib->length;
     uintptr_t ksm_address;
+    uintptr_t ksm_iova;
     size_t ksm_length;
     int list_size;
 
     /* FW requires indirect atomic MR address and length to be aligned
      * to max supported atomic argument size */
     ksm_address = ucs_align_down_pow2((uintptr_t)address,
                                       UCT_IB_MD_MAX_MR_SIZE);
-    ksm_length  = mr_length + (uintptr_t)address - ksm_address;
+    ksm_iova = ucs_align_down_pow2((uintptr_t)iova, UCT_IB_MD_MAX_MR_SIZE);
```

Comment:
`ksm_iova = iova + (ksm_address - address)`

for memic: iova=mmap(...)+offset; address=NULL
for nonmemic: iova=vaddr+offset; address=vaddr

Reply:
Done